### DISCUSSION TITLE
DRM Video Content

### DISCUSSION THREAD

**Discussion by aritworksdev** [CommunityToolkit/Maui#2349] [Q&A]
Can "mediaelement" play DRM content? If so, can someone please give me a sample code.

Thanks
R

**Comment by bijington** [comment]
I don't believe it can. Do you have an example of how you would expect to use it?

**Reply by aritworksdev** [reply]
I do have it working in web version using VideoJS player. But when I use videojs, on native app model (.NET MAUI) it is failing.

That's why I am looking for alternative to use ExoPlayer.

**Reply by aritworksdev** [reply]
It should be something like this

https://docs.axinom.com/services/drm/players/exoplayer

**Reply by JonathanWebsdale** [reply]
Ideally the DRM support would be added as requested for the Xamarin MediaElement back in 2020 (https://github.com/xamarin/XamarinCommunityToolkit/issues/648),  That means supporting widevine on Android, Fairplay on iOS/MacOs, and PlayReady on Windows. Allowing the passing of a set of values for each of the 3 DRM standards and have the video player use those to decode and playback.

API Changes
This will need to be part of changes that are consistent with: https://github.com/xamarin/XamarinCommunityToolkit/issues/646

Intended Use Case
Any company producing content in video format needs DRM video playback.

**Reply by bijington** [reply]
Hey @JonathanWebsdale I know that you are much more knowledgeable on this topic over myself. I see mention in (https://github.com/xamarin/XamarinCommunityToolkit/issues/648) that provides a link on how to achieve this in Windows (UWP). I also see that the link to ExoPlayer shows how to provide support for Widevine, we will be migrating away from using ExoPlayer because support is obsolete. @ne0rrmatrix has started migrating over to Media3 but I believe that migration is blocked right now. 

I love the idea of including DRM support but we would need to work out how to achieve this on each of the platforms once the android migration and .NET 9.0 support is completed

**Reply by ne0rrmatrix** [reply]
Media 3 blocker should be removed on patch Tuesday with Maui update that will include fix which was merged 2 weeks ago. Fingers crossed.

**Comment by JPDelprat** [comment]
Hi,
I really need DRM support for my use case.
Has there been any progress on this topic?

Thank you.
